server/serviceadapter.js

```diff
@@ -0,0 +1,40 @@
+class ServiceAdapter {
+  /**
+   * A string that identifies this service adapter.
+   */
+  get serviceId() {
+    if (this.constructor.SERVICE_ID == null) {
+      throw `Service adapter ${this.constructor.name} does not have a static SERVICE_ID property`;
+    }
+
+    return this.constructor.SERVICE_ID;
+  }
+
+  /**
+   * A boolean that indicates whether video metadata can be safely cached.
+   * @returns {boolean}
+   */
+  get cacheSafe() {
```

Comment:
```suggestion
  get isCacheSafe() {
```

Reply:
I wasn't really sure about this one. I went from `enableCache` to `cacheEnabled` to `cacheSafe` until I decided to leave it as-is and wait for input. I agree that `isCacheSafe` is better, though.